src/main/java/bio/terra/cli/Context.java

```diff
@@ -0,0 +1,207 @@
+package bio.terra.cli;
+
+import bio.terra.cli.exception.SystemException;
+import bio.terra.cli.exception.UserActionableException;
+import bio.terra.cli.serialization.disk.DiskContext;
+import bio.terra.cli.utils.JacksonMapper;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Internal representation of the current context or state. This class maintains singleton instances
+ * of the internal state classes (Config, Server, User, Workspace).
+ */
+public class Context {
+  private static final Logger logger = LoggerFactory.getLogger(Context.class);
+
+  // singleton objects that define the current context or state
+  private static Config currentConfig;
+  private static Server currentServer;
+  private static User currentUser;
+  private static Workspace currentWorkspace;
```

Comment:
Yeah I had this structure in the first version of this PR (`GlobalContext.get()`), so I see where you're coming from, but I don't think it really makes sense here. We'll never have more than one `Context` instance, because it corresponds very closely with what's on disk.
e.g. `User.login` writes user credentials to a Google credential store on disk, and a pet SA key to a json file on disk. So I'm not sure what a call to `User.login` that doesn't change global state would mean.